dulwich/archive.py

```diff
@@ -17,77 +16,87 @@
 """
 
+from contextlib import closing
+from io import BytesIO
+from os import SEEK_END
 import posixpath
 import stat
-import tarfile
 import struct
-from os import SEEK_END
-from io import BytesIO
-from contextlib import closing
+import tarfile
 
 
-class ChunkedBytesIO(object):
+class ChunkedBytesIO(object):  # pylint: disable=too-few-public-methods
     """Turn a list of bytestrings into a file-like object.
 
     This is similar to creating a `BytesIO` from a concatenation of the
     bytestring list, but saves memory by NOT creating one giant bytestring
-    first::
+    first:
 
         BytesIO(b''.join(list_of_bytestrings)) =~= ChunkedBytesIO(
             list_of_bytestrings)
     """
 
     def __init__(self, contents):
+        """Init."""
         self.contents = contents
         self.pos = (0, 0)
 
     def read(self, maxbytes=None):
+        """Read.
+
+        Args:
+            maxbytes: int
+
+        Returns:
+            Bytestrings
+        """
         if maxbytes < 0:
             maxbytes = float("inf")
 
         buf = []
         chunk, cursor = self.pos
 
         while chunk < len(self.contents):
-            if maxbytes < len(self.contents[chunk]) - cursor:
-                buf.append(self.contents[chunk][cursor : cursor + maxbytes])
-                cursor += maxbytes
-                self.pos = (chunk, cursor)
-                break
-            else:
+            if maxbytes >= len(self.contents[chunk]) - cursor:
                 buf.append(self.contents[chunk][cursor:])
                 maxbytes -= len(self.contents[chunk]) - cursor
                 chunk += 1
                 cursor = 0
                 self.pos = (chunk, cursor)
+            else:
+                buf.append(self.contents[chunk][cursor: cursor + maxbytes])
+                cursor += maxbytes
+                self.pos = (chunk, cursor)
+                break
         return b"".join(buf)
 
 
-def tar_stream(store, tree, mtime, prefix=b"", format=""):
+def tar_stream(store, tree, mtime, prefix=b"", compression_format=""):
     """Generate a tar stream for the contents of a Git tree.
 
     Returns a generator that lazily assembles a .tar.gz archive, yielding it in
     pieces (bytestrings). To obtain the complete .tar.gz binary file, simply
     concatenate these chunks.
 
     Args:
-      store: Object store to retrieve objects from
-      tree: Tree object for the tree root
-      mtime: UNIX timestamp that is assigned as the modification time for
-        all files, and the gzip header modification time if format='gz'
-      format: Optional compression format for tarball
+        store: Object store to retrieve objects from
+        tree: Tree object for the tree root
+        mtime: UNIX timestamp that is assigned as the modification time for
+            all files, and the gzip header modification time if
+            compression_format='gz'
+        prefix:
+        compression_format: Optional compression format for tarball
```

Comment:
This is not just the compression format for the tarball, it can also specify e.g. "zip" for non-compression tarballs.

Reply:
Why does it need to be renamed? It causes disruption for callers.

Reply:
I mean what is written in Docs.